docs/proposals/PROPOSAL_E.md

```diff
@@ -72,6 +53,25 @@ Create a new artifact media type to support future use cases where a separate co
 }
 ```
 
+#### Annotations
+
+The following optional annotations should be added to the artifact manifest:
+
+- `org.opencontainers.artifact.type`: type of artifact (sig, sbom, etc)
+- `org.opencontainers.artifact.description`: human readable description for the artifact
+- `org.opencontainers.artifact.created`: creation time for a manifest
+
+Additional annotations should be considered for filtering various artifact types, e.g. signature public key hash, attestation type, and sbom schema.
+
+To maintain backwards compatibility, these annotations should be considered from the current config schema:
+
+- `org.opencontainers.platform.architecture`: CPU architecture for binaries
+- `org.opencontainers.platform.os`: operating system for binaries
+- `org.opencontainers.platform.os.version`: operating system version for binaries
+- `org.opencontainers.platform.variant`: variant of the CPU architecture for binaries
+
+Existing `org.opencontainers.image.*` annotations should be reviewed to consider more generic names (e.g. replacing `image` with `manifest` or `artifact`).
```

Comment:
Annotations apply to both the image-spec and proposed artifact-spec manifests, so we'll want this either above or below both.

Reply:
> It's more confusing to me, but I'm a bit too close to it, so if it's helpful to others I'll defer to them.
> 
> Originally the proposal was ordered in the rollout order, what you can do with image spec and digest tags today, followed by a new artifact spec and an API in the future. That got shifted to prioritize the end state first, which helps show our vision but may confuse people looking for how this will be rolled out.

This is actually why I would like your feedback. I personally may not have understood the order. My proposed order is this: 1. Introduced additions/modifications 2. API contingency for old registries 3. Upgrade path for existing manifests. Does this make sense to you?

Reply:
I think the goal makes sense, but the proposal doesn't do a good job highlighting that. That's part of the challenge of writing one doc and everyone trying to answer a different question when reading it.